pkg/sfu/helpers.go

```diff
@@ -157,6 +157,73 @@ func setVP8TemporalLayer(pl []byte, s *DownTrack) (payload []byte, skip bool) {
 	return
 }
 
+// isH264Keyframe detects if h264 payload is a keyframe
+// this code was taken from https://github.com/jech/galene/blob/codecs/rtpconn/rtpreader.go#L45
+// all credits belongs to Juliusz Chroboczek @jech and the awesome Galene SFU
+func isH264Keyframe(payload []byte) bool {
+	if len(payload) < 1 {
+		return false
+	}
+	nalu := payload[0] & 0x1F
+	if nalu == 0 {
+		// reserved
+		return false
+	} else if nalu <= 23 {
+		// simple NALU
+		return nalu == 5
+	} else if nalu == 24 || nalu == 25 || nalu == 26 || nalu == 27 {
+		// STAP-A, STAP-B, MTAP16 or MTAP24
+		i := 1
+		if nalu == 25 || nalu == 26 || nalu == 27 {
+			// skip DON
+			i += 2
+		}
+		for i < len(payload) {
+			if i+2 > len(payload) {
+				return false
+			}
+			length := uint16(payload[i])<<8 |
+				uint16(payload[i+1])
+			i += 2
+			if i+int(length) > len(payload) {
+				return false
+			}
+			offset := 0
+			if nalu == 26 {
+				offset = 3
+			} else if nalu == 27 {
+				offset = 4
+			}
+			if offset >= int(length) {
+				return false
+			}
+			n := payload[i+offset] & 0x1F
+			if n == 5 {
+				return true
+			} else if n >= 24 {
+				// is this legal?
+				return false
```

Comment:
This line should probably be removed — in Galène, it gives up at this point, but Ion-SFU is different, it should continue walking the STAP.

Reply:
done, thank you!